contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,335 @@
+// Package mgo provides functions and types which allow tracing of the MGO MongoDB client (https://github.com/globalsign/mgo)
+package mgo // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/globalsign/mgo"
+
+import (
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(url string, opts ...MongoOption) (*Session, error) {
+	session, err := mgo.Dial(url)
+	s := &Session{
+		Session: session,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// Session wraps a Session with the Context and Config information
```

Comment:
What is `Context` and `Config`? Keep in mind that this documentation is for users. We shouldn't mention anything implementation related here.

The best way to review your documentation is by changing perspective away from your code editor.  In your shell:
```bash
go get golang.org/x/tools/cmd/godoc
godoc -http=:8080
```
Then open `http://localhost:8080/pkg` and find this package. This is what users will be looking at. Whatever we document here is for this page.

So, maybe we can try something like (inspired from our other contribs):
```go
// Session wraps a msg.Session and automatically traces attached calls.
```
This should be understandable and clear to the user. They can then see all attached methods on the page.